include/picongpu/fields/incidentField/profiles/BaseParam.hpp

```diff
@@ -96,7 +96,35 @@ namespace picongpu
                             return 0.0;
                         }
 
+                        /** SFINAE detection if the user parameter define the variable TIME_DELAY_SI
+                         *
+                         * This allows that time delay can be an optional variable a user must only define if needed.
+                         * The default if it is not defined is 0.
+                         * @{
+                         */
+                        template<typename T, typename = void>
+                        struct GetTimeDelay
+                        {
+                            static constexpr float_X value = 0.0;
+                        };
+
+                        template<typename T>
+                        struct GetTimeDelay<T, decltype((void) T::TIME_DELAY_SI, void())>
+                        {
+                            static constexpr float_X value = T::TIME_DELAY_SI;
+                        };
+
                     public:
+                        /** Time delay
+                         *
+                         * unit: UNIT_TIME
+                         */
+                        static constexpr float_X TIME_DELAY
+                            = static_cast<float_X>(GetTimeDelay<Params>::value / UNIT_TIME);
+                        PMACC_CASSERT_MSG(
+                            _error_laser_time_delay_must_be_positive____check_your_incidentField_param_file,

```

Comment:
Why does it have to be positive? 
Wouldn't a negative delay not just start the init before the simulation (which is not possible) and thus skip till start. 
Might be numerically questionably but valid option for delay scans to include negative values as well. 

This is just a question out of curiosity and does nor prevent merging.

Reply:
The incident field pulses are only initialized on the Huygens-Surfaces, so a negative `TIME_DELAY` could lead to a cut-off laser pulse, because it is (unlike with the fieldBackground approach) not initialized in the simulation volume itself. I don't really see, why you physically would want to have that.